pkg/application/types.go

```diff
@@ -53,3 +56,43 @@ type NodePool struct {
 type DefaultAppsValues struct {
 	BaseDomain string `yaml:"baseDomain"`
 }
+
+// UnmarshalJSON implements a custom unmarshaller that handles both the old and new schema structures
+func (cv *ClusterValues) UnmarshalJSON(b []byte) error {
+	if strings.Contains(string(b), `"global":`) {
```

Comment:
We can also have global in the old schema 😬 Yeah, it's not pretty 😄

Reply:
![tableflip](https://github.com/giantswarm/clustertest/assets/3384072/26f0a434-b856-4b35-b24f-a7a21ca83164)

Reply:
So if the `Metadata` only present in the new schema? 
From your PR:
```
func ClusterIsUsingNewHelmValuesSchema(valuesWithNewSchema *ClusterValues) bool {
	return !reflect.DeepEqual(valuesWithNewSchema.Global.Metadata, Metadata{})
}
```

Reply:
Mine is also incorrect, sorry about that, I had it written a bit differently initially, but it got lost as I was moving code around.